File: Go/K08.go
package main

import (
	"fmt"
	"math/rand"
	"strconv"
)

type K08 struct {}

func (k08 K08) execute(num int) {
	switch num {
	case 1: k08.question1()
	case 2: k08.question2()
	case 3: k08.question3()
	case 4: k08.question4()
	}
}

func (k08 K08) question1() {
	var numbers []int
	var inputString string

	for i := 0; i < 3; i++ {
		fmt.Println(fmt.Sprintf("%dつ目の値を入力してください。＞", i + 1))
		_, _ = fmt.Scanln(&inputString)
		_num, _ := strconv.Atoi(inputString)
		numbers = append(numbers, _num)
	}

	fmt.Println("どちらを調べますか？")
	fmt.Print("（０：最大値　１：最小値）＞")
	_, _ = fmt.Scanln(&inputString)
	choice, _ := strconv.Atoi(inputString)

	switch choice {
	case 0:
		fmt.Println(fmt.Sprintf("%dの中で最大値は%dです。",
			len(numbers), getMaxValue(numbers[0], numbers[1], numbers[2])))
	case 1:
		fmt.Println(fmt.Sprintf("%dの中で最小値は%dです。",
			len(numbers), getMinValue(numbers[0], numbers[1], numbers[2])))
	}
}

func (k08 K08) question2() {
	fmt.Println("冒険が今始まる！")
	playerHp := 200 + rand.Intn(101)
	continueGame := 1
	var inputString string

	for continueGame != 0 {
		if playerHp == 0 {
			return
		}
		fmt.Println(fmt.Sprintf("あなたのHP：%d", playerHp))
		fmt.Print("奥に進みますか？（１：奥に進む　０．帰る）＞")
		_, _ = fmt.Scanln(&inputString)
		continueGame, _ = strconv.Atoi(inputString)
		if continueGame == 1 {
			golemLv := rand.Intn(5)
			attackGolem(golemLv, &playerHp)
		}
	}
	fmt.Println("リレ〇ト！")
}

func (k08 K08) question3() {
	return
}

func (k08 K08) question4() {
	return
}

File: Dart/lib/K08.dart
import 'dart:io';
import 'dart:math';
import 'package:Dart/Executable.dart';
import 'package:Dart/K08_1sub.dart';
import 'package:Dart/K08_2sub.dart';

class K08 extends Executable {
  @override
  void Question1() {
    try {
      List<int> numbers = [];
      for (int i = 0; i < 3; i++) {
        stdout.write('${i + 1}つ目の値を入力してください。＞');
        numbers.add(int.parse(stdin.readLineSync()));
      }

      print('どちらを調べますか？');
      stdout.write('（０：最大値　１：最小値）＞');

      int choice = int.parse(stdin.readLineSync());

      switch (choice) {
        case 0:
          {
            int maxValue = GetMaxValue(numbers[0], numbers[1], numbers[2]);
            print('${numbers.length}の中で最大値は$maxValueです。');
            break;
          }
        case 1:
          {
            int minValue = GetMinValue(numbers[0], numbers[1], numbers[2]);
            print('${numbers.length}の中で最大値は$minValueです。');
            break;
          }
      }
    } on Exception catch (e) {
      print(e.toString());
    }
  }

  @override
  void Question2() {

    print('冒険が今始まる！');
    Random rand = Random();
    int playerHp = 200 + rand.nextInt(101);
    int continueGame = 1;

    try {
      while (continueGame != 0) {
        print('あなたのHP：$playerHp');
        stdout.write('奥に進みますか？（１：奥に進む　０．帰る）＞');
        continueGame = int.parse(stdin.readLineSync());
        if (continueGame >= 1) {
          int golemLv = rand.nextInt(5);
          playerHp = AttackGolem(golemLv, playerHp);
          if (playerHp == 0) {
            break;
          }
        }
      }
      print('リレ〇ト！');
    } on Exception catch (e) {
      print(e.toString());
    }
  }

  @override
  void Question3() {

  }

  @override
  void Question4() {
    // TODO: implement Question4
  }
}